src/core/qgsaction.cpp

```diff
@@ -74,6 +87,183 @@ void QgsAction::run( const QgsExpressionContext &expressionContext ) const
     else
       QDesktopServices::openUrl( QUrl( expandedAction, QUrl::TolerantMode ) );
   }
+  else if ( mType == QgsAction::SubmitUrlEncoded || mType == QgsAction::SubmitUrlMultipart )
+  {
+
+    QUrl url{ expandedAction };
```

Comment:
would be nice to pull all this out to a separate function -- the ::run method is very long as a result!

Reply:
done